src/ParseError.js

```diff
@@ -22,7 +22,14 @@ class ParseError extends Error {
     this.code = code;
     Object.defineProperty(this, 'message', {
       enumerable: true,
-      value: message,
+      value:
+        typeof message === 'string'
+          ? message
+          : typeof message === 'object' &&
+            typeof message.toString === 'function' &&
```

Comment:
Why `typeof message.toString === 'function' &&` and `message.toString() !== '[object Object]'`?

Reply:
yeah that's crazy I know, I'm not sure why yet but inside the test env `.toString()` is giving `[object Object]`.

Reply:
Maybe [util.inspect()](https://nodejs.org/api/util.html#utilinspectobject-options) will give you a more reliable and better formatted output.

Reply:
but isn't `util` package from node? does it work in browser too? cause this ParseError thing is from the js sdk

Reply:
nevermind we really need this check.

answering your question, due to this edge case that might be uncommon but who knows 

![Screenshot 2022-11-26 at 14 55 43](https://user-images.githubusercontent.com/719763/204102558-52204901-cb7f-4b1c-9c0f-3086be96bd4c.png)

that's a case where nobody will be ever able to extract a readable message so Parse would just throw `Parse Error: `  instead of `Parse Error: [object Object]`

Reply:
unless I'm missing something, that's not how Errors should be thrown, but I see lots of `Parse Error: [object Object]` in my server logs and would like to cover this case as well. I've already added a test case exclusive for it, going to push soon.

Reply:
take a look now @mtrezza it's more refined and test covered

Reply:
and this is how it's looking like

![Screenshot 2022-11-26 at 15 58 11](https://user-images.githubusercontent.com/719763/204104898-83c82250-4f22-4036-acde-b81e00c64898.png)

Reply:
> but isn't util package from node? 

yes you are right, sorry for mixing things up here

> due to this edge case 

I think that was my point, we should never just throw an object directly without context. For example throwing `new Error({})` should not make it through review. Looking at the [spec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error), I would say that `new Error(message)` expects a string, not an object.